org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/DebugServer.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import org.eclipse.jdt.ls.core.internal.debug.IDebugServer;
+import org.eclipse.jdt.ls.debug.internal.adapter.DebugSession;
+import org.eclipse.jdt.ls.debug.internal.adapter.DispatcherProtocol;
+import org.eclipse.jdt.ls.debug.internal.adapter.DispatcherProtocol.IResponder;
+import org.eclipse.jdt.ls.debug.internal.adapter.Events.DebugEvent;
+import org.eclipse.jdt.ls.debug.internal.adapter.JsonUtils;
+import org.eclipse.jdt.ls.debug.internal.adapter.Results.DebugResult;
+import org.eclipse.jdt.ls.debug.internal.adapter.Results.ErrorResponseBody;
+import org.eclipse.jdt.ls.debug.internal.adapter.Types.Message;
+import org.eclipse.jdt.ls.debug.internal.core.log.Logger;
+
+import com.google.gson.JsonObject;
+
+public class DebugServer implements IDebugServer {
+	private ServerSocket _serverSocket = null;
+	private DebugSession debugSession = null;
+
+	public DebugServer() {
+		try {
+			_serverSocket = new ServerSocket(0);
+		} catch (IOException e) {
+			e.printStackTrace();

```

Comment:
e.printStackTrace() should not be used and should be replaced with some kind of error handling work, eg : log and wrap it into a new exception.